pkg/generate/git/git.go

```diff
@@ -0,0 +1,97 @@
+package git
+
+import (
+	"bufio"
+	"io"
+	"net/url"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// ParseRepository parses a string that may be in the Git format (git@) or URL format
+// and extracts the appropriate value. Any fragment on the URL is preserved.
+//
+// Protocols returned:
+// - http, https
+// - file
+// - git
+func ParseRepository(s string) (*url.URL, error) {
```

Comment:
@smarterclayton I have feeling that we are doing this on various places in the origin/STI... we should perhaps create some utility package for GIT and re-use this in both project. And since we are using GIT in STI and origin imports STI, I think the right place to have these utility methods would be STI.